packages/babel-plugin-redux-saga/src/index.js

```diff
@@ -142,8 +146,8 @@ module.exports = function(babel) {
     CallExpression(path, state) {
```

Comment:
shouldnt we just visit `yield`s? probably would make reasoning about this easier

Reply:
not sure on which line I should comment about this, but it should be easy to replace those IIFEs with comma separated expressions
```js
var _ref;
yield _ref = call(smthelse), _ref[_SAGA_LOCATION] = { fileName: ..., lineNumber: ... } ,_ref;
```
Can be achieved with things like `scope.generateDeclararedUidIdentifier('ref')` and `path.replaceWithMultiple` - or smth like that 😉

Reply:
@restrry WDYT?

Reply:
make sense, but imho wrapping with `Object.defineProperty` makes code a bit more readable (for debugging, for example), so I'd stick to IIFE.

Reply:
what do u think about this one?

> shouldnt we just visit yields? probably would make reasoning about this easier

@restrry

Reply:
I did it in earlier implementation, but found several places in our codes base when people used expressions, like
```js
function*() {
  yield foo.bar(1, 2, 3) || {}
}
// or
function* () {
  yield (foo? bar() : baz())
}
```
we can try to get rid of expression support, that should simplify code

Reply:
well, idiomatic way of working with sagas is generally yielding effects only which are just objects - so generally it should be quite "safe" to wrap even more complex expressions, as their completion values should be objects, right?

Reply:
actually yes, if we're using Symbol. but in case of using fallback to string key is not so 'safe'. probably, that's a minor concern and we can return to the initial implementation and wrap only yielded values

Reply:
string key will still be used with `defineProperty`, what concern do u see about this in comparison with Symbol situations? Seems to me that both of them are the same regarding "safety" of this

Reply:
totally agree, changed